### ISSUE TITLE
Since Adding Blinds and Reapter (come with the Blinds) constantly "Gateway is not reachable"

### ISSUE THREAD

**Issue by tehmilcho** [treban/pimatic-tradfri#39]
#### Problem / Feature : 
Hey, i dont know why but since i added the Blinds inlc the needed Reapter to my Gateway i have some days i got constantly "gateway is not reachable", i control my Ikea devices over Pimatic and most off them are triggert by Motionsensor (zwave) over Rules. 

I notice this (apart from the Erorrlog in Pimatic) that i get a longer delay between Motion dection and swichting on the Lights or sometimes only one of two groups where switched. 

I also have a connection to the Gateway by Iobroker, so frist i think "okay, iobroker / dual-Connection is the Problem" so i switched off the Module in Iobroker for 3-4 days but its same. 

Im long time User of you Plugin and before the blinds i got the Error 1-3 times a Week but most in the Nighttimes (maybe a reconnect of the Fritzbox or so).

i have at the moment in my Gateway 36 Lights, 5 Blinds, 3 Reapter and 3 Remotes for Blinds.

Troubleshoting so Far:
- Reboot of the Gateway
- Shutdown of the Iobroker Module
- Reboot of Pimatic (helps often a while)

LOG: 
_error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
14:12:07error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
14:11:26error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
14:09:25error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
14:06:25error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
01:46:43error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
01:45:33error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
01:41:41error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
01:40:41error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
00:18:31error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
00:16:21error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
2019-12-16 23:16:40error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
22:48:10error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
19:40:54error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
19:40:54error [pimatic-tradfri]: set device flur-2-1 error: gateway not reachable
19:39:51error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
19:37:01error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
19:35:41error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:42:29error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:40:29error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:34:58error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:34:08error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:31:18error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:29:28error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:28:18error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
17:25:58error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!_

**Comment by treban** [comment]
Okay, I'll try to check it. 
can you please activate the debug mode and post the log when the error occurs.

i'm currently implementing blind support and will check if the state machine can be optimized.

**Comment by tehmilcho** [comment]
Hey, sure i can do it... But has everytime you report something its working in the moment fine. Its so strange i have no ideas what trigger this issue.

It is a problem for performance that i have my groups and the single lights added to pimatic or do it dont care?

**Comment by tehmilcho** [comment]
hey ... today i upgrade to Ubuntu 18.04 LTS and got this problem again .. but with Debug there not really more infos.

debug [pimatic-tradfri]: Light Schlafzimmer-4 is offline
09:40:20debug [pimatic-tradfri]: Light Schlafzimmer-3 is offline
09:40:20debug [pimatic-tradfri]: Light Schlafzimmer-2 is offline
09:40:20debug [pimatic-tradfri]: Obeserving now the device Schlafzimmer Rollo - Links
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-6
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-5
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-4
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-3
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device Wohnzimmer-1
09:40:20debug [pimatic-tradfri]: Obeserving now the group Wohnzimmer Deckenlampe
09:40:20debug [pimatic-tradfri]: Obeserving now the device Schlafzimmer-4
09:40:20debug [pimatic-tradfri]: Obeserving now the device Schlafzimmer-3
09:40:20debug [pimatic-tradfri]: Obeserving now the device Schlafzimmer-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device Schlafzimmer-1
09:40:20debug [pimatic-tradfri]: Obeserving now the group Schlafzimmer Deckenlampe
09:40:20debug [pimatic-tradfri]: Obeserving now the group Abstellkammer
09:40:20debug [pimatic-tradfri]: Obeserving now the group Bar
09:40:20debug [pimatic-tradfri]: Obeserving now the device Bar-3
09:40:20debug [pimatic-tradfri]: Obeserving now the device Bar-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device Bar-1
09:40:20debug [pimatic-tradfri]: Obeserving now the device esstisch-1
09:40:20debug [pimatic-tradfri]: Obeserving now the device esstisch-4
09:40:20debug [pimatic-tradfri]: Obeserving now the device esstisch-3
09:40:20debug [pimatic-tradfri]: Obeserving now the device esstisch-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-2-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-1-4
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-1-2
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-1-3
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-1-1
09:40:20debug [pimatic-tradfri]: Obeserving now the device flur-2-1
09:40:20debug [pimatic-tradfri]: Obeserving now the group Esstisch
09:40:20debug [pimatic-tradfri]: Obeserving now the group Flur2
09:40:20debug [pimatic-tradfri]: Obeserving now the group Flur1
09:40:20debug [pimatic-tradfri]: Tradfri plugin ready
09:40:20debug [pimatic-tradfri]: Gateway online - Firmware: 1.9.27
09:40:20error [pimatic-tradfri]: Try to connect to Tradfri
09:40:20error [pimatic-tradfri]: ...connection error!
09:40:16error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
09:39:30error [pimatic-tradfri]: Try to connect to Tradfri
09:39:30error [pimatic-tradfri]: ...connection error!
09:39:26error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
09:38:00error [pimatic-tradfri]: Try to connect to Tradfri
09:38:00error [pimatic-tradfri]: ...connection error!
09:37:56error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
09:36:50error [pimatic-tradfri]: Try to connect to Tradfri
09:36:50error [pimatic-tradfri]: ...connection error!
09:36:46error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!
09:36:29warn [pimatic-echo]: device with id 25 not found
09:35:50error [pimatic-tradfri]: Try to connect to Tradfri
09:35:50error [pimatic-tradfri]: ...connection error!
09:35:46error [pimatic-tradfri]: Tradfri gateway is not reachable anymore!

i restarted multiple times pimaitc without success .. 50% of the Actions will be failed.... so i rebooted the Gateway ... after this my Pimatic will be floated by Errors from the Plugin .. i have then also to restart pimatic again.

Here there Errors after the Gateway Reboot and before the last Pimatic Reboot:
error [pimatic-tradfri]: Gateway is not reachable!
09:56:21error [pimatic-tradfri]: Try to connect to Tradfri
09:56:21error [pimatic-tradfri]: ...connection error!
09:56:17error [pimatic-tradfri]: { Error: No reply in 6s
    at Timeout.<anonymous> (/home/pi/pimatic-app/node_modules/tradfri-coapdtls/lib/retry_send.js:65:18)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10) retransmitTimeout: 6 }
09:56:17error [pimatic-tradfri]: Gateway is not reachable!
09:56:11error [pimatic-tradfri]: Try to connect to Tradfri
09:56:11error [pimatic-tradfri]: ...connection error!
09:56:07error [pimatic-tradfri]: { Error: No reply in 6s
    at Timeout.<anonymous> (/home/pi/pimatic-app/node_modules/tradfri-coapdtls/lib/retry_send.js:65:18)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10) retransmitTimeout: 6 }

**Comment by tehmilcho** [comment]
Hey,
I found a some similar things when the problem happens.. I use in the two roums ikea lights with pimatic over rules via motion sensors, in both i split the lights in 2 groups because there are 2 lamps with 2-4 lights in it. Every the problem happens it will switch the frist group in my rule and give out the error by the second group. Maybe the gateway has some times a hart cap for query that i now reach? Because iobroker and pimatic grap data? 🤔

**Comment by treban** [comment]
with the plugin i use a queue so that there is only one request at a time. 
maybe parallel requests from different sources can influence the gateway.

maybe you can observe the connection on your test system? with the plugin beta? i made some adjustments there.

**Comment by tehmilcho** [comment]
Hey i got the Problem most after Reboots from the System (-> Iobroker and Pimatic will be rebooted) or after some multiple Reboots of pimatic, Then i got the Problem a while and most after a Gateway and Pimatic reboot it work fine. 

I use the Iobroker Connection to Gateway at monent for the Alexa Skill to get the Blinds in it because pimatic hasnt support it. Maybe it my problem is gone when the Plugin support the Blinds because than i use the IoBroker Plugin for Pimatic to get all Pimatic Devices to Iobroker and from Iobroker to the Iobroker Alexa Skill.

**Comment by tehmilcho** [comment]
I hope I don't get on your nerves.. :-)... I started a new test because it happens frequently again... I now disabled iobroker connection to the gateway, reboot the gateway and pimatic after the gateway wars started.

Im troubleshooting i find out that only to restart pimatic helps just a couple hours or reduced the errors. 

The only reason for me to use the tradfi Puglin in iobroker wars to get the blinds in the alexa skill from iobroker. 

I changed it now by adding a dummy-dim and a rule to change both blinds at a time (beause groups are not working in the alpha / beta) and added this dummys over the iobroker pimatic plugin to the skill.

I hope that i  can now get rid of the errors and delays.
